.github/workflows/_rust.yml

```diff
@@ -64,6 +64,79 @@ jobs:
       - uses: ./.github/actions/setup-rust
       - run: cargo test --all-features ${{ matrix.packages }}
 
+  temp-build-windows-artifacts:
+    runs-on: windows-2019
+    defaults:
+      run:
+        working-directory: ./rust
+    strategy:
+      fail-fast: false
+      # The matrix is 1x1 to match the style of build-push-linux-release-artifacts
+      # In the future we could try to cross-compile aarch64-windows here.
+      matrix:
+        name:
+          - package: hello-world
+            artifact: hello-world
+    env:
+      BINARY_DEST_PATH: ${{ matrix.name.artifact }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-rust
+        with:
+          targets: x86_64-pc-windows-msvc
+      - name: Build release binaries
+        run: |
+          cargo build --release -p ${{ matrix.name.package }}
+          cp target/release/${{ matrix.name.package }}.exe "${{ env.BINARY_DEST_PATH }}-x64.exe"
+          pwd
+          ls *.exe
+      - name: Save build artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.name.artifact }}-x64.exe
+          path: ${{ github.workspace }}/rust/${{ env.BINARY_DEST_PATH }}-x64.exe
+
+  # This should be identical to `build-push-windows-release-artifacts` in `cd.yml` except for the permissions, needs, and uploading step
+  temp-build-tauri:
+    runs-on: windows-2019
+    defaults:
+      run:
+        working-directory: ./rust
+    strategy:
+      fail-fast: false
+      # The matrix is 1x1 to match the style of build-push-linux-release-artifacts
+      # In the future we could try to cross-compile aarch64-windows here.
+      matrix:
+        name:
+          - package: firezone-windows-client
+            artifact: windows-client
+    env:
+      BINARY_DEST_PATH: ${{ matrix.name.artifact }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: ./.github/actions/setup-rust
+        with:
+          targets: x86_64-pc-windows-msvc
+      - name: Build release binaries
+        run: |
+          cargo install tauri-cli
```

Comment:
If this step takes any length of time, would it be worth adding it as a cargo.toml build dependency so that it's cached?

Reply:
It does take a super long time. `cargo install` invokes `cargo build` internally so if sccache will pick up anything that uses `cargo build` maybe it'll get picked up anyway? cargo-tauri is supposed to end up in PATH so I'm not sure if adding it as a build dependency would work. I don't think the getting started docs mentioned it.

You said sccache caches anything built on main, so the only reason I'm still seeing slow builds is that the running the workflow on a PR can't update sccache, right? It treats the cache as read-only?

Reply:
Ah, I think you're right.